fast_llm/layers/transformer/config.py

```diff
@@ -244,6 +250,12 @@ def _validate(self) -> None:
                 )
 
 
+for name in PeftType:
+    # We need this because we are using the reserved field name `type`.
+    # TODO: Implement proper dynamic typing.
+    TransformerPeftConfig.register_subclass(name.value, TransformerPeftConfig)
+
```

Comment:
can we use `tag` instead of `type` to avoid this workaround?

Reply:
We could but I don't think it's worth it. The workaround won't stay long since I already have dynamic implementations in #245. Also keeping `type` makes the transition easier since we're already keeping that name, eg. so config change or backward compatibility fix is needed.